From 7e411375119482b92e3bfd09d49499d09b710ed8 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 10 Jul 2018 21:32:07 +0200
Subject: [PATCH 2/2] Updater: Improve battery check

* Lower default value by 10%
* Decrease the level by another 10% if charging
* Always allow installation in case device doesn't have a battery

Change-Id: Iebb10220612006fbd096eb474cf3034ef52144b3
---
 res/values/integers.xml                           |  3 ++-
 src/org/lineageos/updater/UpdatesListAdapter.java | 11 ++++++++++-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/res/values/integers.xml b/res/values/integers.xml
index 5b68147..9cb9e2d 100644
--- a/res/values/integers.xml
+++ b/res/values/integers.xml
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <integer name="battery_ok_percentage">40</integer>
+    <integer name="battery_ok_percentage_charging">20</integer>
+    <integer name="battery_ok_percentage_discharging">30</integer>
 </resources>
diff --git a/src/org/lineageos/updater/UpdatesListAdapter.java b/src/org/lineageos/updater/UpdatesListAdapter.java
index 5766a35..7186b62 100644
--- a/src/org/lineageos/updater/UpdatesListAdapter.java
+++ b/src/org/lineageos/updater/UpdatesListAdapter.java
@@ -540,6 +540,15 @@ public class UpdatesListAdapter extends RecyclerView.Adapter<UpdatesListAdapter.
     private boolean isBatteryLevelOk() {
         BatteryManager bm = mActivity.getSystemService(BatteryManager.class);
         int percent = bm.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY);
-        return percent >= mActivity.getResources().getInteger(R.integer.battery_ok_percentage);
+        int status = bm.getIntProperty(BatteryManager.BATTERY_PROPERTY_STATUS);
+        switch (status) {
+            case BatteryManager.BATTERY_STATUS_FULL:
+            case BatteryManager.BATTERY_STATUS_UNKNOWN:
+                return true;
+            case BatteryManager.BATTERY_STATUS_CHARGING:
+                return percent >= mActivity.getResources().getInteger(R.integer.battery_ok_percentage_charging);
+            default:
+                return percent >= mActivity.getResources().getInteger(R.integer.battery_ok_percentage_discharging);
+        }
     }
 }
-- 
2.17.1

